components/clp-package-utils/clp_package_utils/scripts/native/decompress.py

```diff
@@ -26,6 +38,124 @@
 logger.addHandler(logging_console_handler)
 
 
+def get_orig_file_id(db_config: Database, path: str) -> Optional[str]:
+    """
+    Returns the original file id of the file with the given path
+    If multiple files have the same path, this method returns
+    the first file returned from the archive metadata database
+    :param db_config:
+    :param path:
+    :return: The original file id of a file with the given path.
+    If no file matches with input path, returns None
+    """
+    sql_adapter = SQL_Adapter(db_config)
+    with closing(sql_adapter.create_connection(True)) as db_conn, closing(
+        db_conn.cursor(dictionary=True)
+    ) as db_cursor:
+        db_cursor.execute(
+            f"SELECT orig_file_id FROM `{CLP_METADATA_TABLE_PREFIX}files` WHERE path = (%s)",
+            (path,),
+        )
+        results = db_cursor.fetchall()
+        db_conn.commit()
+
+        if len(results) == 0:
+            logger.error("No file found for the given file path.")
+            return None
+
+        if len(results) > 1:
+            logger.warning(
+                "Multiple files found for the given file path, "
+                "return the orig_file_id of the first file"
+            )
+
+        return results[0]["orig_file_id"]
+
+
+def submit_and_monitor_ir_extraction_job_in_db(
+    db_config: Database,
+    orig_file_id: str,
+    msg_ix: int,
+    target_uncompressed_size: int | None,
```

Comment:
The syntax is only valid Python 3.10 and above, which is not by default available on Ubuntu Focal (3.8) and Jammy (3.9).
```suggestion
    target_uncompressed_size: Optional[int],
```